### ISSUE TITLE
Rule proposal: prefer EventTarget instead of EventEmitter

### ISSUE THREAD

**Issue by jimmywarting** [sindresorhus/eslint-plugin-unicorn#1740]
### Description

why? `EventTarget` exist in Deno and browser (unlike `EventEmitter` - which is a NodeJS only thing) and `EventTarget` too also has support for using the options `{ once, signal }` to cancel some event listener you can also know if something is "isTrusted" and emitted by nodejs itself. It also have other useful tool such stopPropagation, target, timestamp and other things. maybe we will expect to see more use of it when/if nodejs decides to add support for something like `ExtendableEvent.waitUntil`

using this rule could also potentially reduce the bundle size and make you think of other solutions that is more cross env friendlier. sometimes you might not always need a event handler at all and you can simply just pass options to a constructor or something else

### Fail

```js
import EventEmitter from 'node:event'

class Foo extends EventEmitter { ... }
```

```js
foo.on('ready', fn)
```

### Pass

```js
class Foo extends EventTarget { ... }
```
```js
// ✓ ok (most oftan you only need one listener)

class Foo {
  onmessage = null
  
  init() {
    this.ws.onmessage = ({data}) => {
      doSomethingWith(data)
      this.onmessage?.(data)
    }
  }
}
```
```js
foo.ready.then(fn)
```

**Comment by fregante** [comment]
Great, I didn't know EventTarget was available in node https://nodejs.org/api/events.html#eventtarget-and-event-api

**Comment by sindresorhus** [comment]
Accepted
